src/util.js

```diff
@@ -8,7 +8,11 @@ const resolver = require('./resolver')
 exports = module.exports
 
 exports.serialize = (dagNode, callback) => {
-  callback(null, cbor.encode(dagNode))
+  try {
+    callback(null, cbor.encode(dagNode))
```

Comment:
calling the `callback` inside the `try catch` block might backfire because from then on, any following code that gets executed and throws, will come back to this `try catch` block.

Best pattern is:

```
let serialized 
try {
  serialized = cbor.encode(dagNode)
} catch (err) {
  // return here is also important, otherwise in case of error the execution would continue
  return callback(err)
}
callback(null, serialized)
```